src/nbl/asset/utils/CHLSLCompiler.cpp

```diff
@@ -61,19 +71,71 @@ CHLSLCompiler::~CHLSLCompiler()
 }
 
 
-struct DxcCompilationResult
+static void try_upgrade_hlsl_version(std::vector<std::wstring>& arguments) 
 {
-    ComPtr<IDxcBlobEncoding> errorMessages;
-    ComPtr<IDxcBlob> objectBlob;
-    ComPtr<IDxcResult> compileResult;
+    auto foundShaderStageArgument = std::find(arguments.begin(), arguments.end(), L"-HV");
+    if (foundShaderStageArgument != arguments.end()) {
+        auto foundShaderStageArgumentValueIdx = foundShaderStageArgument - arguments.begin() + 1;
+        std::wstring version = arguments[foundShaderStageArgumentValueIdx];
+        if (version.length() >= 4 && (version[2] < '2' || (version[2] == '2' && version[3] <= '0'))) 
+            arguments[foundShaderStageArgumentValueIdx] = L"2021";
+    }
+}
 
-    std::string GetErrorMessagesString()
+
+static void try_upgrade_shader_stage(std::vector<std::wstring> &arguments) {
+    auto foundShaderStageArgument = std::find(arguments.begin(), arguments.end(), L"-T");
+    if (foundShaderStageArgument != arguments.end()) {
+        auto foundShaderStageArgumentValueIdx = foundShaderStageArgument - arguments.begin() + 1;
+        std::wstring targetProfile = arguments[foundShaderStageArgumentValueIdx];
+        if (targetProfile.length() >= 6) {
+            int argument_version = (targetProfile[3] - '0') * 10 + (targetProfile[5] - '0');
+            if (argument_version < 67)
```

Comment:
don't check length, find the SECOND `_` from the back, take the suffix, replace the last `_` with a `.`, then parse as two ints (version), finally do your upgrades and re-assembly

the way you wrote it now will break if we ever get SM 6.10 (they're shipping 6.8 now)